.pre-commit-config.yaml

```diff
@@ -0,0 +1,17 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+    - id: check-added-large-files
+    - id: mixed-line-ending
+      args: ['--fix', 'no']
+      exclude: |
+         (?x)(
+            .*\.cmd
+         )$
+    - id: mixed-line-ending
+      args: ['--fix', 'crlf']
+      files: '\.cmd$'
```

Comment:
Can we achieve the same all via .gitattributes? We can add there exclusions, default conversion, etc.

Reply:
Thanks, I agree that doing one single big commit that fixes the line endings in all text files (all but MD, JPG, PNG, etc), and then supplemented with correct .gitattributes to ensure no wrong line endings will be checked in again via some PR.

Reply:
Hi @MariusNi I have added rules. Pleas recheck.  even if the commits will be squashed as one big commit I tried to show nice git history where each time I added rule, the next commit is adding all files changed by invoking `pre-commit run --all-files` . As this can be a lot I think looking at individual  commits and how rule affect the existing code would be good way to review it. Eg
rule
https://github.com/Azure/azure-osconfig/pull/810/commits/e3a025ce06b873f7ac5636baf01216896ce23499
Rule execution
https://github.com/Azure/azure-osconfig/pull/810/commits/b64b5479e5be9c72a1c981d6dace5959c66f7268
and next commits after those.
As this is a lot hope that helps to see if those rules are the way we want to uphold.
I also changed logic from exclude to explicit allow where rule runs